core/src/main/java/org/apache/iceberg/hadoop/Util.java

```diff
@@ -84,10 +88,38 @@ public static String[] blockLocations(FileIO io, ScanTaskGroup<?> taskGroup) {
     return locations.toArray(HadoopInputFile.NO_LOCATION_PREFERENCE);
   }
 
+  public static boolean usesHadoopFileIO(FileIO io, String location) {
+    if (io instanceof HadoopFileIO) {
+      return true;
+
+    } else if (io instanceof ResolvingFileIO) {
+      ResolvingFileIO resolvingFileIO = (ResolvingFileIO) io;
+      return resolvingFileIO.ioClass(location).isAssignableFrom(HadoopFileIO.class);
```

Comment:
Shouldn't this be the other way around?

```
HadoopFileIO.class.isAssignableFrom(resolvingFileIO.ioClass(location));
```
https://stackoverflow.com/a/6983596/4108401